CalPatRec/src/TZClusterFinder_module.cc

```diff
@@ -583,52 +598,47 @@ namespace mu2e {
   //-----------------------------------------------------------------------------
   void TZClusterFinder::combineChunks() {
 
-    _f.biggestChi2combine = 1000.;
+    double smallestDeltaTime = 10000.;
+    int    foundValidCombine = 0;
+    size_t chunkOneIdx       = 0;
+    size_t chunkTwoIdx       = 0;
 
-    // set seed chunk
     for (size_t i=0; i<_f.chunks.size()-1; i++) {
-      // add seed to fitter
-      _f.chi2seed = _f.chunks[i].fitter.chi2Dof();
-      _f.seedTime = _f.chunks[i].avgTime;
+      if (_f.chunks[i].nrgSelection != _f.chunks[i+1].nrgSelection) {continue;}
       _f.seedZpos = _f.chunks[i].avgZpos;
-      // now set test chunk
-      for (size_t k=i+1; k<_f.chunks.size(); k++) {
-        if (_f.chunks[i].nrgSelection != _f.chunks[k].nrgSelection) {continue;}
-        _f.testTime = _f.chunks[k].avgTime;
-        _f.testZpos = _f.chunks[k].avgZpos;
-        if (std::abs(_f.seedTime-_f.testTime) >= _combineWindow) {continue;}
-        if (_f.chunks[i].nCombines == 0 && _f.chunks[k].nCombines == 0
-            && std::abs(_f.seedZpos-_f.testZpos) > _maxCombineSep) {continue;}
-        // add test to fitter
-        _f.chunks[i].fitter.addSum(_f.chunks[k].fitter);
-        _f.chi2combineTest = _f.chunks[i].fitter.chi2Dof();
-        if (_f.chi2combineTest < _chi2combineThresh) {
-          if (_f.chi2combineTest < _f.biggestChi2combine) {
-            _f.biggestChi2combine = _f.chi2combineTest;
-            _f.chunkOneIdx = i;
-            _f.chunkTwoIdx = k;
-          }
-        }
-        // remove test from fitter
-        _f.chunks[i].fitter.removeSum(_f.chunks[k].fitter);
+      _f.testZpos = _f.chunks[i+1].avgZpos;
+      if (_f.chunks[i].nCombines == 0 && _f.chunks[i+1].nCombines == 0
+          && std::abs(_f.seedZpos-_f.testZpos) > _maxCombineSep) {continue;}
+      _f.seedTime = _f.chunks[i].avgTime;
+      _f.testTime = _f.chunks[i+1].avgTime;
+      double deltaTime = std::abs(_f.seedTime - _f.testTime);
+      if (deltaTime > _combineWindow) {
+        if (foundValidCombine == 1) {break;}
+        else {continue;}
+      }
+      foundValidCombine = 1;
+      if (deltaTime < smallestDeltaTime) {
+        smallestDeltaTime = deltaTime;
+        chunkOneIdx = i;
+        chunkTwoIdx = i+1;
       }
     }
 
-    if (_f.biggestChi2combine < 1000.) {
-      for (size_t i=0; i<_f.chunks[_f.chunkTwoIdx].hIndices.size(); i++) {
-        _f.chunks[_f.chunkOneIdx].hIndices.push_back(_f.chunks[_f.chunkTwoIdx].hIndices[i]);
+    if (smallestDeltaTime < 10000.) {
```

Comment:
instead of using a constant (`10000.`), can you create a `static` variable outside the loop (which you can also use to initialize the variable `smallestDeltaTime`?

Reply:
If you decide to keep it initialized in code, and not a fcl parameter, I second Giani's request but with one small change.  Instead of static, make it constexpr.  This gives the compiler and optimizer the most information and gives the optimizer permission to do all optimizations.  I can elaborate if you want.

I also second Dave's request that if you just need a big number, use std::numeric_limits.

Reply:
i do just need a big number ... so i will use std::numeric_limits. thanks for the input!